torch/quantization/_equalize.py

```diff
@@ -0,0 +1,89 @@
+import torch
+import torch.nn as nn
+
+def channel_range(tensor, axis =0):
+    ''' finds the range of weights associated with a specific channel
+
+    '''
+    size_of_tensor_dim = len(tensor.size())
+    mins = tensor
+    maxs = tensor
+    for i in range(size_of_tensor_dim - 1, axis, -1):
+        mins = torch.min(mins, i)[0]
+        maxs = torch.max(maxs, i)[0]
+    for i in range(axis):
+        mins = torch.min(mins, 0)[0]
+        maxs = torch.max(maxs, 0)[0]
+    return maxs.sub(mins)
+
+def scaling_channels(tensor1, tensor2, output_axis = 0, input_axis = 1):
+    ''' Given two adjacent tensors', the weights are scaled such that
+    the ranges of the first tensors' output channel are equal to the
+    ranges of the second tensors' input channel
+
+    Note: assumes the input/output channels are the first two axises of the given tensors
+    '''
+    if tensor1.size()[output_axis] != tensor2.size()[input_axis]:
+        raise TypeError("Incompatible tensors")
+
+    output_channel_tensor1 = channel_range(tensor1, output_axis)
+    input_channel_tensor2 = channel_range(tensor2, input_axis)
+
+    # producing scaling factors to applied
+    zero_input = torch.zeros(output_channel_tensor1.size())
+    scaling_factors = torch.sqrt(torch.addcdiv(zero_input, output_channel_tensor1, input_channel_tensor2))
+    r_scaling_factors = torch.reciprocal(scaling_factors)
+
+    # formatting the scaling (1D) tensors to be applied on the given arguement tensors
+    scaling_factors = torch.unsqueeze(scaling_factors, output_axis)
+    r_scaling_factors = torch.unsqueeze(r_scaling_factors, input_axis)
+
+    for i in range(len(tensor1.size()) - 2):
```

Comment:
Can you explain the purpose of the loop here?

Reply:
this loop is for when dim of a tensor is greater than 2, when dealing with a conv2d we get a 4d tensor and I was trying to pad the scaling_factors tensors with the extra dims so i could then multiply in the next few lines (broadcasting didn't work for me when I tried the first time, can give it another go)

Reply:
original scaling factor is one number, I think it might be clearer to do something like this:
1. get a list of 1s with the same dimension as original tensor, e.g. sizes = [1] * tensor1.dim()
2. set particular axis to scaling factor,  sizes[out_axis] = scaling_factor